Neos.Neos/Classes/FrontendRouting/Projection/DocumentTypeClassification.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\Neos\FrontendRouting\Projection;
+
+use Neos\ContentRepository\Core\NodeType\NodeTypeManager;
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+use Neos\Neos\Domain\Service\NodeTypeNameFactory;
+
+enum DocumentTypeClassification: string
```

Comment:
ALERT: Naming discussion incoming :see_no_evil: 

I think that `DocumentTypeClassification` does not reflect the intended semantics well. Instead, I think that `RoutingStrategy` may be a better term.

What I have in mind is very close to the semantics presented here, but with a key difference. `RoutingStrategy` would have 5 cases as well:

```php
enum RoutingStrategy
{
    case STRATEGY_DOCUMENT;
    case STRATEGY_SHORTCUT;
    case STRATEGY_SITE;
    case STRATEGY_NONE;
    case STRATEGY_PLACEHOLDER; // PLACEHOLDER is explicit
}
```

I believe this would make the purpose of this enum much clearer on the consuming side.

Reply:
I would also volunteer making the adjustments on a nested pr to `5077-documentUriPlaceholders` ;)

Reply:
Well, imho this comes down to reasoning and decision and which part we put in code and which in comments.
Basically we have the phrase "I have a document of type X and thus decide to use strategy y".
The current alternatives are that we either model the `DocumentTypeClassification` and comment the decision we make (status quo) or we model the `RoutingStrategy` and comment the reasoning (the suggested change). I guess it's up to perspective at this point.
What we of course could do is model both to reduce/avoid commenting. WDYT?

Reply:
I think an interesting question to ask here is if any of this might become an extension point?

Reply:
Okay i see but regardless of the decision i like to come to a solution fast as this is the last blocker for beta11!! ;)

Reply:
We've encountered a similar conundrum in the UI before. In Neos.Neos.Ui I once introduced the (unfortunate) concept of "Node Type Roles" to be able to distinguish between `Document`s, `Content`s and `ContentCollection`s without hard-coding their Node Type names. These concepts all imply different behavior.

If we were to add `Shortcut`s and `Site`s to that list, I think what we'd be looking at would be an enumeration of all domain-specific terms that are introduced by Neos.Neos on top of the ContentRepository without having any explicit representation. (* well, `Site`s do have a representation, but that is besides the point)

`DocumentTypeClassification` comes close to an explicit model in that sense, but is only introduced in service to this particular projection and only covers documents (which of course makes sense in the context of this projection).

@nezaniel 
> Basically we have the phrase "I have a document of type X and thus decide to use strategy y".
> [...]
> What we of course could do is model both to reduce/avoid commenting. WDYT?

I agree. Additionally, I would argue that the suggested `RoutingStrategy` model should stay close to the projection, while `DocumentTypeClassification` may be turned into a broader concept covering all of those Neos domain-specifics and can live at a higher level.

@kitsunet 
> And with that in mind, shouldn't the RoutingStrategies if we named them so be independently named from the document (type) to make it clear the rather are there for doing a specific type of routing and not FOR a specfici document type?

That's a good point. I totally agree.